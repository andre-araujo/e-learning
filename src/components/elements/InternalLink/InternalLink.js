import React from 'react';
import { node, number, string } from 'prop-types';
import Link from 'next/link';

import {
  StyledAnchor,
} from './InternalLink.styles';

function InternalLink({
  children,
  href,
  gap,
  ...props
}) {
  return (
    <Link href={href}>
      <StyledAnchor gap={gap} {...props}>
        {children}
      </StyledAnchor>
    </Link>
  );
}

InternalLink.defaultProps = {
  children: '',
  href: null,
  gap: 0,
};

InternalLink.propTypes = {
  children: node,
  href: string,
  gap: number,
};

export default InternalLink;
