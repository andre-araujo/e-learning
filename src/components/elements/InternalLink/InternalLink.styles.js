import styled, { css } from 'styled-components';

export const StyledAnchor = styled.a`
  cursor: pointer;
  text-decoration: underline;
  user-select: none;
  display: inline-block;
  font-size: 0.8em;

  :hover {
    color: orange;
  }

  ${({ gap }) => gap && css`
    margin-right: ${gap / 2}px;

    @media (min-width: 640px) {
      margin-right: ${gap}px;
    }
  `};

  ${({ margin }) => margin && css`
    margin: ${margin};
  `};

  @media (min-width: 640px) {
    font-size: 1em;
  }
`;
