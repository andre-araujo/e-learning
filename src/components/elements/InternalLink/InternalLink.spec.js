import React from 'react';
import { mount } from 'enzyme';

import InternalLink from './InternalLink';

describe('InternalLink component', () => {
  it('should match snapshot', () => {
    expect(mount(<InternalLink />)).toMatchSnapshot();
  });
});
