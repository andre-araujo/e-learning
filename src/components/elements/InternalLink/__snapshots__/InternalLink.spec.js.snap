// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink component should match snapshot 1`] = `
<styled.a
  gap={0}
  href="#"
/>
`;

exports[`InternalLink component should match snapshot with anchor 1`] = `
<styled.a
  gap={0}
  href="#"
>
  content
</styled.a>
`;

exports[`InternalLink component should match snapshot with nect link 1`] = `
<Link
  href="/test"
>
  <styled.a
    gap={0}
  >
    content
  </styled.a>
</Link>
`;
